lambdas/common/src/person-identity-service.ts

```diff
@@ -0,0 +1,50 @@
+import { DynamoDBClient, QueryCommand } from "@aws-sdk/client-dynamodb";
+import { unmarshall } from "@aws-sdk/util-dynamodb";
+import { withRetry } from "./util/retry";
+import { PersonIdentityItem } from "./types/person-identity";
+import { isRecordExpired } from "./util/is-record-expired";
+import { RecordExpiredError, RecordNotFoundError } from "./exceptions/errors";
+
+const personIdentityTableName = `person-identity-${process.env.CommonStackName}`;
+
+export async function getPersonIdentity(
+  sessionId: string,
+  /** Optional parameter; used for mocking the DynamoDB client when testing. */
+  dynamoClient: DynamoDBClient = new DynamoDBClient()
+) {
+  async function queryPersonIdentity() {
+    const command = new QueryCommand({
+      TableName: personIdentityTableName,
+      KeyConditionExpression: "sessionId = :value",
+      ExpressionAttributeValues: {
+        ":value": {
+          S: sessionId,
+        },
+      },
+    });
+
+    const result = await dynamoClient.send(command);
+
+    if (result.Count !== 1 || !result.Items) {
```

Comment:
Should be `result.Count == 0` - I've seen scenarios in the code that cover "more than 1 session record" and so you don't want to throw `RecordNotFoundError` because could have been found.

Reply:
Okay, makes sense. I realise I don't think this condition is covered in unit tests so I'll also add a test for it. In the case that there are multiple records for the same session id, is there any rule we should follow for choosing one (eg, most recently created / last expiry) or is the current `result.Items[0]` sufficient?

Reply:
Should be okay for now, I think we can think about this when we have a common-ts lib